### PR TITLE
clone presentation materials

### PR DESCRIPTION
**Pull request by romanetar**
ref https://tipit.avaza.com/project/view#!tab=task-pane&groupby=MyTaskDueDate&view=vertical&task=3393805&stafilter=NotComplete&fileview=grid

Signed-off-by: romanetar <roman_ag@hotmail.com>

### CODE REVIEW

File: app/Models/Foundation/Summit/Events/Presentations/Materials/PresentationMaterial.php [hunk 1]
```diff
@@ -200,4 +200,45 @@ public function clearPresentation(){
     }
 
     use ScheduleEntity;
+
+    public function clone(): PresentationMaterial{
```
Comment (smarcet) [inline app/Models/Foundation/Summit/Events/Presentations/Materials/PresentationMaterial.php:204]: this clone method should not be resolved statically on the base class
each class should implement its clone method and resolve it through polimorphism

File: app/Models/Foundation/Summit/Events/Presentations/Presentation.php [hunk 2]
```diff
@@ -894,6 +894,18 @@ public function setMaterials($materials)
         $this->materials = $materials;
     }
 
+    /**
+     * @param PresentationMaterial $material
+     * @return Presentation
+     */
+    public function addMaterial(PresentationMaterial $material)
```
Comment (smarcet) [inline app/Models/Foundation/Summit/Events/Presentations/Presentation.php:901]: use this implementation here
https://github.com/OpenStackweb/summit-api/blob/ebda2ab8320efd613b70b1b7397ef31559f3c0d1/app/Models/Foundation/Summit/Events/Presentations/Presentation.php#L522
https://github.com/OpenStackweb/summit-api/blob/ebda2ab8320efd613b70b1b7397ef31559f3c0d1/app/Models/Foundation/Summit/Events/Presentations/Presentation.php#L652
https://github.com/OpenStackweb/summit-api/blob/ebda2ab8320efd613b70b1b7397ef31559f3c0d1/app/Models/Foundation/Summit/Events/Presentations/Presentation.php#L686
https://github.com/OpenStackweb/summit-api/blob/ebda2ab8320efd613b70b1b7397ef31559f3c0d1/app/Models/Foundation/Summit/Events/Presentations/Presentation.php#L735

**Review by smarcet — CHANGES_REQUESTED** [review]
@romanetar please review commnets

File: app/Models/Foundation/Summit/Events/Presentations/Presentation.php (hunk 2 above)
Comment (romanetar) [reply app/Models/Foundation/Summit/Events/Presentations/Presentation.php:901]: done

File: app/Models/Foundation/Summit/Events/Presentations/Materials/PresentationMaterial.php (hunk 1 above)
Comment (romanetar) [reply app/Models/Foundation/Summit/Events/Presentations/Materials/PresentationMaterial.php]: done

**Review by smarcet — APPROVED** [review]
LGTM

**Pull request merged**